Adds a CLI that automatically generates C++ code

MichaelKim20 opened the pull request:
Adds a CLI that automatically generates C++ code that calculates the offset of the virtual method on the C++ side.
1. I added a CLI that automatically generates C++ code.
2. I applied the generated C++ code.

The previous PR is #689 #710

Relates to #323

AndrejMitrovic commented on source/agora/cli/gen/main.d:
```diff
@@ -0,0 +1,132 @@
+    foreach (member; __traits(allMembers, scpd.scp.SCPDriver.SCPDriver))
+    {
+        if ((member == "__dtor") || (member == "__xdtor") || (member == "ValidationLevel")) continue;
+        methods ~= member;
```
This code is too specialized for SCPDriver, and `methods` may include fields here and not just methods, because there is no check that the member is a method.

AndrejMitrovic commented on source/scpp/extra/DVTableChecks.cpp:
```diff
@@ -0,0 +1,199 @@
+
+int checkOrderOfVM ()
+{
+    auto df0 = &SCPDriverDlang::signEnvelope;
```
This is just comparing two C++ classes. It doesn't check the bindings. It's not what the task was about.

MichaelKim20 committed:
Find offset of virtual destructor on C++ side - extern (C++)

MichaelKim20 committed:
Check the address of the destructor - extern (C++)

MichaelKim20 committed:
Check the offset of virtual methods - extern (C++)

MichaelKim20 committed:
Check that the offset of the virtual method for any class - extern (C++)

MichaelKim20 committed:
Add CLI to automatically generate the source - extern (C++)

This generates a source that calculates
the offset of the virtual method on the C++ side.

MichaelKim20 committed:
Apply automatically generated source from `agora-gen` - extern (C++)
